src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6972,16 +6998,34 @@ private void VisitUsingVariableDeclarationOperation(IUsingDeclarationOperation o
             FinishVisitingStatement(operation);
             _currentStatement = saveCurrentStatement;
         }
-#nullable disable
 
-        public IOperation Visit(IOperation operation)
+        public IOperation? Visit(IOperation? operation)
         {
             // We should never be revisiting nodes we've already visited, and we don't set SemanticModel in this builder.
-            Debug.Assert(operation == null || ((Operation)operation).OwningSemanticModel.Compilation == _compilation);
+            Debug.Assert(operation == null || ((Operation)operation).OwningSemanticModel!.Compilation == _compilation);
             return Visit(operation, argument: null);
         }
 
-        public override IOperation Visit(IOperation operation, int? argument)
+        [return: NotNullIfNotNull("operation")]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public IOperation? VisitRequired(IOperation? operation, int? argument = null)
```

Comment:
📝 I'm not a fan of this. The only reason this method is needed is `OperationVisitor<TArgument, TResult>.DefaultVisit` is `virtual` instead of `abstract`. I would prefer to see a derived class `OperationVisitorWithResult<TArgument, TResult>` derived from the base `OperationVisitor<TArgument, TResult>`, with `DefaultVisit` marked both `override` and `abstract` and all the return types changed from `TResult?` to `TResult`.

Reply:
> The only reason this method is needed is OperationVisitor<TArgument, TResult>.DefaultVisit is virtual instead of abstract

No. Many of the methods in the builder actually can and do return null: anything returning a statement will do so. This is the reality of how this class actually functions.